### PR TITLE
fix: ensure capacity hiccup

### PR DESCRIPTION
**Pull request by BrianAmadori**
## What does this PR change?

This PR attempts to fix the hiccups caused by `ComputeBoneWeights` function when an avatar is loaded.

Before:
image

After:
image

Note that:
* Most of the overhead the `List<T>` capacity setting went away due to use of arrays
* Very high GC overhead almost was almost wiped
* In the shown case, the ms cost went from 4.51ms to 0.10ms with the improvement enabled

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarMeshCombiner/AvatarMeshCombinerUtils.cs [hunk 1]
```diff
@@ -42,11 +42,14 @@ public static class AvatarMeshCombinerUtils
...
         /// <returns>A list of BoneWeights that share the same skeleton.</returns>
-        public static List<BoneWeight> ComputeBoneWeights( List<CombineLayer> layers )
+        public static BoneWeight[] ComputeBoneWeights( List<CombineLayer> layers )
         {
-            List<BoneWeight> result = new List<BoneWeight>();
             int layersCount = layers.Count;
 
+            int resultSize = 0;
+
+            List<BoneWeight[]> boneWeightArrays = new List<BoneWeight[]>(10);
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarMeshCombiner/AvatarMeshCombinerUtils.cs:51]: where is this `10` coming from?

**Review by AjimenezDCL — APPROVED** [review]
Impresive, great job!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarMeshCombiner/AvatarMeshCombinerUtils.cs (hunk 1 above)
Comment (BrianAmadori) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarMeshCombiner/AvatarMeshCombinerUtils.cs:51]: just guessing the max possible wearable parts that might be coming

**Review by sandrade-dcl — APPROVED** [review]
Good improvement, good work!

**Pull request merged**